src/config/localfile-config.c

```diff
@@ -634,51 +643,88 @@ int Read_Localfile(XML_NODE node, void *d1, __attribute__((unused)) void *d2)
-            log_config->globs[gl].gfiles->file = NULL;
+                    file++;
+                }
 
-            /* Wildcard exclusion, check for date */
-            if (logf[pl].exclude && strchr(logf[pl].exclude, '%')) {
+                file = 0;
+           
+                while(file < totalFiles) {
 
-                time_t l_time = time(0);
-                char excluded_path_date[PATH_MAX] = {0};
-                size_t ret;
-                struct tm tm_result = { .tm_sec = 0 };
+                    os_free(logf[pl].file);
+                    os_strdup(expand_files[file], logf[pl].file);
+
+                    os_realloc(log_config->globs, (gl + 2)*sizeof(logreader_glob), log_config->globs);
+                    os_strdup(logf[pl].file, log_config->globs[gl].gpath);
+
+
+                    memset(&log_config->globs[gl + 1], 0, sizeof(logreader_glob));
+                    os_calloc(1, sizeof(logreader), log_config->globs[gl].gfiles);
+
+                    memcpy(log_config->globs[gl].gfiles, &logf[pl], sizeof(logreader));
+                    log_config->globs[gl].gfiles->file = NULL;
+
+                    /* Wildcard exclusion, check for date */
+                    if (logf[pl].exclude && strchr(logf[pl].exclude, '%')) {
+
+                        time_t l_time = time(0);
+                        char excluded_path_date[PATH_MAX] = {0};
+                        size_t ret;
+                        struct tm tm_result = { .tm_sec = 0 };
+
+                        localtime_r(&l_time, &tm_result);
+                        ret = strftime(excluded_path_date, PATH_MAX, logf[pl].exclude, &tm_result);
+                        if (ret != 0) {
+                            os_strdup(excluded_path_date, log_config->globs[gl].exclude_path);
+                        }
+                    }
+                    else if (logf[pl].exclude) {
+                        os_strdup(logf[pl].exclude, log_config->globs[gl].exclude_path);
+                    }
+
+                    gl++;
+                    file++;
 
-                localtime_r(&l_time, &tm_result);
-                ret = strftime(excluded_path_date, PATH_MAX, logf[pl].exclude, &tm_result);
-                if (ret != 0) {
-                    os_strdup(excluded_path_date, log_config->globs[gl].exclude_path);
                 }
-            }
-            else if (logf[pl].exclude) {
-                os_strdup(logf[pl].exclude, log_config->globs[gl].exclude_path);
+
+                os_free(expand_files);
+                os_free(result);
             }
 
             if (Remove_Localfile(&logf, pl, 0, 0,NULL)) {
                 merror(REM_ERROR, logf[pl].file);
-                FindClose(hFind);
                 return (OS_INVALID);
-            }
-            log_config->config = logf;
-            FindClose(hFind);
+            } 
+
+             log_config->config = logf;
+
             return 0;
+
```

Comment:
Please remove additional line breaks added to the file.

Reply:
done